gdx/src/com/badlogic/gdx/graphics/g2d/PixmapPacker.java

```diff
@@ -326,66 +322,61 @@ public synchronized void dispose () {
-			if (page.texture == null) {
-				if (page.rects.size != 0 && page.addedRects.size > 0) {
-					page.texture = new Texture(new PixmapTextureData(page.image, page.image.getFormat(), useMipMaps, false, true)) {
-						@Override
-						public void dispose () {
-							super.dispose();
-							getTextureData().consumePixmap().dispose();
-						}
-					};
-					page.texture.setFilter(minFilter, magFilter);
-
-					for (String name : page.addedRects) {
-						Rectangle rect = page.rects.get(name);
-						TextureRegion region = new TextureRegion(page.texture, (int)rect.x, (int)rect.y, (int)rect.width,
-							(int)rect.height);
-						atlas.addRegion(name, region);
-					}
-					page.addedRects.clear();
-				}
-			} else {
-				if (page.addedRects.size > 0) {
+			if (page.addedRects.size > 0) {
+				if (page.texture == null) {
+					generatePageTexture(page, minFilter, magFilter, useMipMaps);
+				} else {
 					page.texture.load(page.texture.getTextureData());
-					for (String name : page.addedRects) {
-						Rectangle rect = page.rects.get(name);
-						TextureRegion region = new TextureRegion(page.texture, (int)rect.x, (int)rect.y, (int)rect.width,
+				}
+				for (String name : page.addedRects) {
+					Rectangle rect = page.rects.get(name);
+					TextureRegion region = new TextureRegion(page.texture, (int)rect.x, (int)rect.y, (int)rect.width,
 							(int)rect.height);
-						atlas.addRegion(name, region);
-					}
-					page.addedRects.clear();
+					atlas.addRegion(name, region);
 				}
+				page.addedRects.clear();
+			}
+		}
+	}
+
+	private void generatePageTexture(final Page page, TextureFilter minFilter, TextureFilter magFilter, boolean useMipMaps) {
+		page.texture = new Texture(new PixmapTextureData(page.image, page.image.getFormat(), useMipMaps, false, true)) {
+			@Override
+			public void dispose() {
+				super.dispose();
+				getTextureData().consumePixmap().dispose();
+			}
+		};
+		page.texture.setFilter(minFilter, magFilter);
+	}
+
+	/** Return an ordered array of texture regions corresponding with the pages of the packer, creating new textures
+	 * if necessary.  The same array will be returned from successive calls to this method, possibly modified with
+	 * additional texture regions if more pages have been added since the last call.  Once this method has been called,
+	 * you must dispose of the resulting texture regions (or equivalently, any atlas generated or updated by this packer)
+	 * yourself - disposing the PixmapPacker will no longer dispose these textures or their backing pixmaps.
+	 */
+	public Array<TextureRegion> getTextureRegions(TextureFilter minFilter, TextureFilter magFilter, boolean useMipMaps) {
+		while (textureRegions.size < pages.size) {
+			if (disposed) return null;
+			Page page = pages.get(textureRegions.size);
+			if (page.texture == null) {
+				generatePageTexture(page, minFilter, magFilter, useMipMaps);
 			}
+			textureRegions.add(new TextureRegion(page.texture));
 		}
+		return textureRegions;

```

Comment:
I like that change, although I can think of one gotcha.  Assuming the new `updateTextureRegions` clears `addedRects` on any pages after doing `texture.load()`, calling it would mean that a subsequent call to `updateTextureAtlas` wouldn't get those rects added.  That would work fine in the way that the FreeTypeFontGenerator is using it, since it doesn't rely on an atlas containing named regions, but might bite someone.

If you're happy to have that gotcha in there, I'll make that change (remove `getTextureRegions`, add `updateTextureRegions`), and make sure to call out the danger in the javadoc of the new method.